src/ServiceManagement/Management/Management.Tests/SubscriptionPrincipalTest.cs

```diff
@@ -0,0 +1,95 @@
+﻿//
+// Copyright (c) Microsoft.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.WindowsAzure.Testing;
+
+namespace Microsoft.WindowsAzure.Management.Testing
+{
+    using System;
+    using System.Linq;
+    using Management;
+    using Management.Models;
+    using Microsoft.Azure.Test;
+    using Xunit;
+    public class SubscriptionPrincipalTest : TestBase
+    {
+        [Fact]
+        public void SubscriptionCRUD()
+        {
+            using (var undoContext = UndoContext.Current)
+            {
+                undoContext.Start();
+                var mgmt = this.GetManagementClient();
+
+                // Retrive existing SPN's if any
+                var existingSPNs = mgmt.SubscriptionServicePrincipals.List();
+
+                // Random SPN Object Id
+                string newSPNObjectId = "5d825630-ab8f-4579-8719-8399b692609d";
```

Comment:
I think hard coding one is for what it is worth now, but just FYI there may be some methods that can help someone who is to update the test code waive re-writing this ID, like using a conditional check to determine using a new ID or re-using an existing one, etc.